industrial_robot_client/src/joint_trajectory_action.cpp

```diff
@@ -216,9 +210,12 @@ void JointTrajectoryAction::cancelCB(JointTractoryActionServer::GoalHandle & gh)
 
 void JointTrajectoryAction::controllerStateCB(const control_msgs::FollowJointTrajectoryFeedbackConstPtr &msg)
 {
-  //ROS_DEBUG("Checking controller state feedback");
+  ROS_DEBUG("Checking controller state feedback");
   last_trajectory_state_ = msg;
-  trajectory_state_recvd_ = true;
+  controller_alive_ = true;
+
+  watchdog_timer_.stop();
+  watchdog_timer_.start();
```

Comment:
afaict, it should be possible to 'restart' a timer using `setPeriod(..)` (eventually in [TimerManager](http://docs.ros.org/indigo/api/roscpp/html/timer__manager_8h_source.html#l00420)). That would remove the need for the `stop()`, `start()` pair.